src/sentry/services/hybrid_cloud/project_key/impl.py

```diff
@@ -31,3 +34,21 @@ def get_project_key_by_region(
         self, *, region_name: str, project_id: str, role: ProjectKeyRole
     ) -> Optional[RpcProjectKey]:
         return self._get_project_key(project_id=project_id, role=role)
+
+    def get_docsupport_data(
+        self, *, region_name: str, organization_ids: List[int], role: ProjectKeyRole
+    ) -> List[Tuple[int, RpcProject, RpcProjectKey]]:
+        project_keys = ProjectKey.objects.filter(
+            status=ProjectKeyStatus.ACTIVE,
+            roles=F("roles").bitor(role.as_orm_role()),
+            project__organization_id__in=organization_ids,
+            project__status=ObjectStatus.ACTIVE,
+        ).select_related("project")
```

Comment:
Should we have an upper limit to the number of records we're willing to fetch here? For large orgs this could be thousands of records which isn't going to be helpful in the docs/develop sites.

Reply:
Hmm, interesting. This should be mirroring [the existing behavior from `get_dsns`](https://github.com/getsentry/getsentry/blob/39651d4/getsentry/web/docsupport.py#L22-L34), meaning that the back end would already be loading and returning thousands of records in those cases. It looks like `DocSupportEndpoint` doesn't paginate them, but maybe the front end truncates or paginates the list.

I'll look into this further.